cpp/include/cudf/reduction.hpp

```diff
@@ -87,6 +86,21 @@ std::unique_ptr<scalar> reduce(
  * @param[in] mr Device memory resource used to allocate the returned scalar's device memory
  * @returns unique pointer to new output column
  */
+std::unique_ptr<column> scan(
+  std::unique_ptr<aggregation> const& agg,
+  scan_type inclusive,
+  null_policy null_handling           = null_policy::EXCLUDE,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Comment:
I thought on this some more and wonder why the "order by" column can't be used as the values to scan/groupby scan? Then we don't need a separate overload.

Reply:
And if there is more than one order by column, then we could use a struct for it.

Reply:
Did a full refactor around this, seems to be a cleaner integration. There is some type support (list column support and nested struct support) and probably some Java API improvements (grouped table scan with multiple `onColumn` arguments) that we can add as follow on improvments.